controllers/metal3.io/baremetalhost_controller.go

```diff
@@ -227,10 +227,30 @@ func (r *BareMetalHostReconciler) Reconcile(ctx context.Context, request ctrl.Re
 		return ctrl.Result{Requeue: true, RequeueAfter: provisionerNotReadyRetryDelay}, nil
 	}
 
+	// If the unmanaged annotation is set we remove any host from the
+	// provisioner and take no further action
+	if annotations != nil {
+		if _, ok := annotations[metal3v1alpha1.UnmanagedAnnotation]; ok {
+			provResult, err := prov.Delete()
+			if err != nil {
+				return ctrl.Result{}, errors.Wrap(err, "failed to delete on unmanage")
+			}
+			if provResult.Dirty {
+				return ctrl.Result{Requeue: true, RequeueAfter: provResult.RequeueAfter}, nil
+			}
+			info.host.SetOperationalStatus(metal3v1alpha1.OperationalStatusUnmanaged)
```

Comment:
I think we also need some code to set the operational status back to ok (or error, or potentially even discovering) once the annotation is removed.

Reply:
@zaneb Doesn't this have the same problem as tracking the previous state in the `Unmanaged` case?  E.g we don't know what the previous operational status was unless it's saved somewhere, so the only option is to set it back to OK, then expect subsequent reconciliation to modify the status afterwards?

Reply:
I was thinking that OperationalStatus will be set by the state machine when we remove the unmanaged annotation, then revisit whatever state-handler aligns with the current state which will remove the unamanaged status and restore the state-appropriate value (or error)?

Reply:
Ok I see now that not every state handler sets the operational status, so we either have to change that, or just set the status to OK on removal of the annotation AFAICS?

Reply:
Ok I set it back to OK in the latest version